src/rp2/plugin/report/abstract_ods_generator.py

```diff
@@ -158,7 +159,15 @@ def _apply_style_to_cell(sheet: Any, row_index: int, column_index: int, style_na
 
     @classmethod
     def _fill_cell(
-        cls, sheet: Any, row_index: int, column_index: int, value: Any, visual_style: str = "transparent", data_style: str = "default", apply_style: bool = True
+        cls,
+        sheet: Any,
+        row_index: int,
+        column_index: int,
+        value: Any,
+        visual_style: str = "transparent",
+        data_style: str = "default",
+        apply_style: bool = True,
+        currency_code: Optional[str] = None,
```

Comment:
This seems like a lot of repeated code, though.

It would also make the full report and open positions report US specific and need to be classified as such since they output only in USD. It was my understanding that since they were general reports they should handle all currencies.

I understand the need for country specific plugins to be independent, but shouldn't the general plugins be made universal?

Reply:
There shouldn't be any duplication of code (common code can be factored in superclasses): there will be multiple templates, with slight differences (e.g. in the fiat style), but that is the price to pay because the ezodf library doesn't allow programmatic creation of styles. In any event templates won't have to be duplicated if they are exactly the same: it's possible to have symbolic links to a template (a .txt file containing the path to another template). 

It's true however that we may not be able to keep common templates and plugins (they were an initial attempt before we had support for another country): it's OK to move everything to country-specific packages.

Let's clarify with an example. Suppose you want to have two version of rp2_full_report for Japan, one denominated in JPY and one in USD:
* add two Japan-specific templates:
  * plugin/report/data/jp/template_rp2_full_report_jpy.ods: this is a copy of the existing rp2_full_report template but it has fiat styles redefined to use JPY
  * plugin/report/data/jp/template_rp2_full_report_usd.txt: this is a pointer to the existing rp2_full_report template which has fiat defined to USD, so there is no duplication (note that this is not an ods file, but a txt file)
* have two report generator plugins:
  * plugin/report/jp/rp2_full_report_jpy.py and plugin/report/jp/rp2_full_report_usd.py: both of these plugin can be subclasses of the existing rp2_full_report.py plugin (if necessary we can refactor its code into an abstract superclass and have all 3 plugins be sublasses of this new superclass)
  
Additionally we will probably need to move the existing "generic" templates and plugin to the us package. Hope this helps.

Reply:
I think I don't need to subclass. For some reason, I thought this was going to be much more difficult than it actually was. I did have to put templates in both countries so, the full report is now technically country-specific. I didn't touch the open positions. It's not all that useful to me at the moment since I already have an app that does it much more smoothly.

Reply:
The open position shows up in git as modified, though: if there are no important changes, could you revert it to the original?

Reply:
Also let's remove `currency_code`.